pkg/cache/v3/linear.go

```diff
@@ -165,6 +165,41 @@ func (cache *LinearCache) DeleteResource(name string) error {
 	return nil
 }
 
+// SetResources replaces current resources with a new set of resources
+// This function is useful for use cases when DiscoveryRequest#resourceNames is empty.
+// This way watches that follow all resources are triggered only once regardless of how many resources are changed.
+func (cache *LinearCache) SetResources(resources map[string]types.Resource) {
+	cache.mu.Lock()
+	defer cache.mu.Unlock()
+
+	cache.version += 1
+
+	modified := map[string]struct{}{}
+	for name := range cache.resources {
```

Comment:
Some comments would be make this easier to follow, like `// First account for removed resources.` and a similar one for the other for loop